Detectors/TPC/reconstruction/include/TPCReconstruction/KrCluster.h

```diff
@@ -22,33 +22,37 @@ namespace tpc
 
 struct KrCluster {
  public:
-  float totCharge = 0; ///< Total charge of the cluster (ADC counts)
-  float maxCharge = 0; ///< Maximum charge of the cluster (ADC counts)
-  int size = 0;        ///< Size of the cluster (TPCDigits)
-  float meanPad = 0;   ///< Center of gravity (Pad number)
-  float meanRow = 0;   ///< Center of gravity (Row number)
-  float meanTime = 0;  ///< Center of gravity (Time)
-  float sigmaPad = 0;  ///< RMS of cluster in pad direction
-  float sigmaRow = 0;  ///< RMS of cluster in row direction
-  float sigmaTime = 0; ///< RMS of cluster in time direction
-  int sector = 0;      ///< Sector number
+  int size = 0;         ///< Size of the cluster (TPCDigits)
+  int sector = 0;       ///< Sector number
+  int maxChargePad = 0; ///< Pad with max. charge in cluster (for leader pad method)
+  int maxChargeRow = 0; ///< Row with max. charge in cluster (for leader pad method)
```

Comment:
I had to go back to ints, since we are calculating e.g. meanPad. This certainly exceeds the max. value of an unsigned char.
If chars are needed here, I need to re-implement a few things.

Reply:
Hi @HauenderPhilip ,
no sure I understand. unsigned char is 0-255, the max pad number is 137, max row is 151, max sector is 35, max size might be larger for the IROC if really all digits in the search window are above threshold, which I think is not too probable.

Reply:
Hi @wiechula ,
thanks for your reply. In the function "updateTempCluster", the following calculation is performed:
`mTempCluster.meanPad += tempPad * tempCharge;` (similar for row and time)
Later (in the function "updateTempClusterFinal"), it is divided by the total charge to calculate the mean:
`mTempCluster.meanPad *= oneOverQtot;`
However, the variable has an overflow before, since we were summing up `tempPad*tempCharge` which is greater than 255 in many cases.

Reply:
But meanPad is float. maxChargePad/Row is for the leader pad, where no average is built, or?

Reply:
Ehhm..yes. Sorry, stupid mistake, I correct it directly.